stable/aws-load-balancer-controller/templates/service.yaml

```diff
@@ -1,13 +1,31 @@
+---
 apiVersion: v1
 kind: Service
 metadata:
   name: {{ template "aws-load-balancer-controller.namePrefix" . }}-webhook-service
   namespace: {{ .Release.Namespace }}
   labels:
-{{ include "aws-load-balancer-controller.labels" . | indent 4 }}
+    app.kubernetes.io/component: webhook-service
+    {{ include "aws-load-balancer-controller.labels" . | nindent 4 }}
 spec:
   ports:
   - port: 443
     targetPort: webhook-server
   selector:
     {{- include "aws-load-balancer-controller.selectorLabels" . | nindent 4 }}
+---
+apiVersion: v1
```

Comment:
have not kept the service creation conditional since

- it seemed non-intuitive to control it using `serviceMonitor.enabled`
- introducing an additional flag like `metricsService.enabled` meant that both `metricsService.enabled` and `serviceMonitor.enabled` need to be set to enable the monitoring.

Alternative proposal:

- deprecate the value `metricsBindAddr`
- introduce a new block in the values
```
metrics:
  enabled: false # only if enabled will the flag be passed to the controller in the deployment and service be created
  port: 8080
  serviceMonitor:
    enabled: false
    interval: "10s"
    selector: {}
      # prometheus: kube-prometheus
    path: /metrics
```
- Major upgrade in the helm chart version

please share your views

Reply:
alternative solution implemented in #504